ansible_collections/arista/avd/roles/eos_designs/templates/network_services/router-ospf.j2

```diff
@@ -0,0 +1,59 @@
+{# Rendering tenant OSPF #}
+{% for tenant in network_services_data.tenants %}
+{%     for vrf in tenant.vrfs %}
+{%         if vrf.ospf is arista.avd.defined and inventory_hostname in vrf.ospf.nodes | arista.avd.default([inventory_hostname]) %}
+{%             set network_services_data.ospf_configured = true %}
+{%         endif %}
+{%     endfor %}
+{% endfor %}
+{% if network_services_data.ospf_configured is arista.avd.defined(true) %}
+router_ospf:
+  process_ids:
+{%     for tenant in network_services_data.tenants %}
+{%         for vrf in tenant.vrfs %}
+{%             if vrf.ospf is arista.avd.defined %}
+{%                 if vrf.ospf.nodes is not arista.avd.defined or inventory_hostname in vrf.ospf.nodes %}
+{%                     set vrf_ospf_interfaces = [] %}
+{%                     set ospf_process_id = vrf.ospf.process_id | arista.avd.default(vrf.vrf_vni, vrf.vrf_rt, vrf.vrf_id) %}
+{%                     set ospf_router_id = vrf.ospf.router_id | arista.avd.default(switch.router_id) %}
+{%                     for l3_interface in vrf.l3_interfaces | arista.avd.natural_sort %}
+{%                         if l3_interface.nodes is arista.avd.defined and inventory_hostname in l3_interface.nodes and l3_interface.interfaces is arista.avd.defined %}
+{%                             for node in l3_interface.nodes %}
+{%                                 if node == inventory_hostname and l3_interface.ospf.enabled is arista.avd.defined(true) %}
```

Comment:
do the check for `ospf.enabled` first - in the same line where you check if `inventory_hostname` is in `nodes`

Reply:
Fixed.